notecard/main.go

```diff
@@ -101,8 +101,9 @@ func main() {
 	card, err := setupNotecard(transport)
 	if err != nil {
 		log.Printf("while setting up notecard: %v", err)
+	} else {
+		defer card.Close()
```

Comment:
Don't we need this for whenever the service terminates, to safely close down the communication with the notecard?

Reply:
yes, but only if the setup succeeded. If it didn't, then card is nil, and all you get is a panic while trying to Close on a nil.